src/data/token.ts

```diff
@@ -31,26 +34,97 @@ const getBalanceForTokenAndDomain = async (
   return new BigNumber(rewardsPotTotal.toString(10));
 };
 
-const getEthplorerTokenData = async (address: string): Promise<TokenInfo> => {
-  // eslint-disable-next-line max-len, prettier/prettier
-  const endpoint = `//api.ethplorer.io/getTokenInfo/${address}?apiKey=${process
-    .env.ETHPLORER_API_KEY || 'freekey'}`;
-  const response = await fetch(endpoint);
-  const data = await response.json();
-  if (data.error) {
-    throw new Error(`Ethplorer error: ${data.error.message}`);
+const getTokenData = async (
+  {
+    colonyManager,
+    client,
+  }: {
+    colonyManager: ContextType['colonyManager'];
+    client: ApolloClient<object>;
+  },
+  address: Address,
+) => {
+  const tokenAddress = address === '0x0' ? ZERO_ADDRESS : address;
+
+  if (!isAddress(tokenAddress)) {
+    // don't bother looking it up if it's an invalid token address
+    throw Error('Invalid token address');
   }
-  const { name, symbol, decimals } = data;
-  const tokenDetails = {
-    name,
-    symbol,
-    decimals: parseInt(decimals, 10),
-    verified: true,
+
+  // If we're asking for ETH, just return static data
+  if (tokenAddress === ZERO_ADDRESS) {
+    return {
+      __typename: 'Token',
+      verified: true,
+      ...ETHER_INFO,
+    };
+  }
+
+  const tokenClient = await colonyManager.getTokenClient(tokenAddress);
+  const chainData: TokenInfo = await tokenClient.getTokenInfo.call();
+
+  let serverDataResult;
+  try {
+    const { data } = await client.query({
```

Comment:
This query always fails on my machine claiming it can't the `iconHash` prop on some `undefined` object:

![Screenshot from 2020-01-22 17-09-53](https://user-images.githubusercontent.com/1193222/72906041-384a1c80-3d3a-11ea-92a9-795aa6341dce.png)

Reply:
Thanks, this is a good find!